pkg/client/client.go

```diff
@@ -231,7 +239,7 @@ func NewClient(opts ...Option) (*Client, error) {
 	}
 
 	if client.imageFetcher == nil {
-		client.imageFetcher = image.NewFetcher(client.logger, client.docker, image.WithRegistryMirrors(client.registryMirrors), image.WithKeychain(client.keychain))
+		client.imageFetcher = image.NewFetcher(client.logger, client.docker, image.WithRegistryMirrors(client.registryMirrors), image.WithKeychain(client.keychain), image.WithInsecureRegistries(client.insecureRegistries))
```

Comment:
I think there is a misunderstood with this code, for what I can see, `client.insecureRegistries` is always going to be empty. because [this](https://github.com/buildpacks/pack/pull/2077/files#diff-2895f30116602d8fe6b0545c186d4f81247fa4eaab049dab174c6c91c0036e08R192) option is never called when we initialized the `client`.

Usually, the client is being [initialized](https://github.com/buildpacks/pack/blob/main/cmd/cmd.go#L141-L151) with configuration we save in the pack configuration file, see `pack config` command. In this case, the `insecure-registries` are configured at runtime when the user invokes the `pack build` or `pack rebase` commands, but at that point in time, the client was already initialized and the image fetcher was already created. 

I think we need a way to passthrough the `insecure-registries` to the `Fetch` method at runtime

Reply:
@prashantrewar Do you have a chance to take a look at this?

Reply:
I apologize @jjbustamante, right now I am busy with something, on the weekend I will try to fix it.